### DISCUSSION TITLE
インストールしたHiCproで発生するエラーについて

### DISCUSSION THREAD

**Discussion by shohei020212** [nig-sc/nigsc_homepage2#322] [Q&A] [answered]
いつもお世話になっております。

インストールしたHiCproを実行したところ、以下のエラーが発生し、止まってしまいます。

```
/lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found 
/lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found
```

実際にバージョンを確認したところ、`GLIBCXX_3.4.20'はありませんでした。

`strings /usr/lib64/libstdc++.so.6 | grep GLIBCXX_
GLIBCXX_3.4
GLIBCXX_3.4.1
GLIBCXX_3.4.2
GLIBCXX_3.4.3
GLIBCXX_3.4.4
GLIBCXX_3.4.5
GLIBCXX_3.4.6
GLIBCXX_3.4.7
GLIBCXX_3.4.8
GLIBCXX_3.4.9
GLIBCXX_3.4.10
GLIBCXX_3.4.11
GLIBCXX_3.4.12
GLIBCXX_3.4.13
GLIBCXX_3.4.14
GLIBCXX_3.4.15
GLIBCXX_3.4.16
GLIBCXX_3.4.17
GLIBCXX_3.4.18
GLIBCXX_3.4.19
GLIBCXX_DEBUG_MESSAGE_LENGTH`

libstdc++.so.6のアップデートが必要だと思うですが、パスワードがわからないためsudo yum update等もできません。

もし解決方法がございましたら、ご教示いただけますでしょうか。
よろしくお願い致します．

**Comment by oogasawa** [comment]
glibcのバージョンが問題になっているので、HiCproはCentOS7が使っているLinux kernelのバージョンでは使えないというのが答えとなると思います。同様の症状はAsperaでも起こっています。こちらの解決方法もご参照ください。https://github.com/nig-sc/nigsc_homepage2/discussions/290

今のところの解決方法は、Apptainer (旧Singularity）コンテナイメージを使うことです。
既に /usr/local/biotoolsの下にコンテナイメージがあるのではないかと思います。

**Comment by sc-fs** [comment]
/usr/local/biotoolsの下にコンテナイメージを検索しましたが、対応するイメージはありませんでした。
ご指摘のエラーはHiCproのヘルプ表示ではなく、実行時に発生しているかと思われますが、その時のログもしくは画面のはお持ちでは無いですか。

**Reply by shohei020212** [reply]
ご回答いただきありがとうございます。

HiCproのログでは以下のエラーが出ております。
ただ、何が問題になっているのかわからない状況です。

```
Run HiC-Pro 3.0.0
--------------------------------------------
Sun Apr 23 16:25:56 JST 2023
Bowtie2 alignment step1 ...
Logs: logs/P592_01_1_1_DKDL220010716-1A_HJFKKCCX2_L5/mapping_step1.log

--------------------------------------------
Sun Apr 23 17:56:00 JST 2023
Bowtie2 alignment step2 ...
Logs: logs/P592_01_1_1_DKDL220010716-1A_HJFKKCCX2_L5/mapping_step2.log
Exit: Error in reads alignment - Exit
make: *** [/home/shohei-noda/tools/HiC-Pro_3.0.0/bin/../scripts//Makefile:101: bowtie_local] Error 1
```

**Reply by sc-fs** [reply, marked as answer]
ログを参照すると確かに、「****/cutsite_trimming: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found」となっていました。
git展開された物をそのまま利用されているのではないでしょうか。
本体のHiC-Proをコンパイルしていただけますか。
git展開フォルダにあるconfig-install.txtのPREFIX =にインストール先を記載して

```
make configure
make install
```

でコンパイルとインストールができると思います。こちらでインストールした物を使ってもらえないでしょうか

**Reply by shohei020212** [reply]
ありがとうございます。
HiC-Proをコンパイルして、再度動かしてみます。
